File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> ans=new ArrayList();
        int max=candies[0]; 
        int n=candies.length;
        for(int i=1;i<n;i++){ //finding max value in candies
            if(candies[i]>max){
                max=candies[i];
            }
        }
        for(int i=0;i<n;i++){ //checking with extra candies
            if(candies[i]+extraCandies<max){
                ans.add(false);
            }
            else if(candies[i]+extraCandies>=max){
                ans.add(true);
            }

        }
        return ans;

    }
}

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        list=[]
        a=0
        maxcandy=max(candies)
        for i in range(len(candies)):
            a=candies[i]+extraCandies
            if a>=maxcandy:
                list.append(True)
            else:
                list.append(False)
        return list

        